taichi/transforms/insert_scratch_pad.cpp

```diff
@@ -82,14 +82,17 @@ class BLSAnalysis : public BasicStmtVisitor {
       }
       bool matching_indices = true;
       std::vector<std::pair<int, int>> offsets;
+      std::vector<int> coeffs;
       offsets.resize(ptr->indices.size());
+      coeffs.resize(ptr->indices.size());
       int num_indices = (int)ptr->indices.size();
       for (int i = 0; i < num_indices; i++) {
         auto diff = irpass::analysis::value_diff_loop_index(ptr->indices[i],
                                                             for_stmt, i);
         if (diff.linear_related()) {
           offsets[i].first = diff.low;
           offsets[i].second = diff.high;
+          coeffs[i] = diff.coeff;
```

Comment:
Yeah, we should support the `coeff != 1` case here! Maybe we need to check other usages of `linear_related()` if they can also support the `coeff != 1` case?